pkg/client/client.go

```diff
@@ -439,6 +439,31 @@ func (c *KpmClient) CompileTarPkg(tarPath string, opts *opt.CompileOptions) (*kc
 	return c.CompileWithOpts(opts)
 }
 
+// CompileGitPkg will compile the kcl package from the git url.
+func (c *KpmClient) CompileGitPkg(gitUrl, version string, opts *opt.CompileOptions) (*kcl.KCLResultList, error) {
```

Comment:
The parameters of this method are adjusted to look like below, because according to the feedback of our previous code during being used by users, this way of passing parameters directly is a bit poor in scalability.

```
func (c *KpmClient) CompileGitPkg(gitOpts *CloneOption, compileOpts *opt.CompileOptions)
```

Reply:
done!